### PR TITLE
Fix 401 error (ASF) and skip SciHub

### PR DESCRIPTION
**Pull request by SteffanDavies**
Fixed an error where downloading from ASF would cause a 401 authentication error.

Added option to skip SciHub and force ASF client.

### CODE REVIEW

File: eof/download.py [hunk 1]
```diff
@@ -73,14 +73,15 @@ def download_eofs(orbit_dts=None, missions=None, sentinel_file=None, save_dir=".
...
-    if client.server_is_up():
+    if client.server_is_up() and not force_asf:
         # try to search on scihub
         if sentinel_file:
             query = client.query_orbit_for_product(sentinel_file, orbit_type=orbit_type)
         else:
             query = client.query_orbit_by_dt(orbit_dts, missions, orbit_type=orbit_type)
 
         if query:
+            print("Attempting download from SciHub")
```
Comment (scottstanie) [inline eof/download.py:84]:
```suggestion
            logger.info("Attempting download from SciHub")
```

File: eof/download.py [hunk 2]
```diff
@@ -89,13 +90,14 @@ def download_eofs(orbit_dts=None, missions=None, sentinel_file=None, save_dir=".
...
     if not scihub_successful:
+        print("Attempting download from ASF")
         logger.warning("Scihub failed, trying ASF")
         asfclient = ASFClient()
         urls = asfclient.get_download_urls(orbit_dts, missions, orbit_type=orbit_type)
         # Download and save all links in parallel
         pool = ThreadPool(processes=MAX_WORKERS)
         result_url_dict = {
-            pool.apply_async(_download_and_write, (url,)): url
+            pool.apply_async(_download_and_write, args=[url, save_dir, user, password]): url
```
[the remark this answers is not in the document]
Comment (scottstanie) [reply eof/download.py:100]: looks like `user/password` aren't defined here

**Review by scottstanie — COMMENTED** [review]
Thanks for the addition!
Question on the username/password- is this auto-grabbed from someone's `.netrc` file? Or would someone have to manually specify in python (and therefore it wouldn't work currently from the command line)?

if it needs manual entry, maybe we could add something like what `asfsmd` has to get the auth: https://github.com/avalentino/asfsmd/blob/b959bd38b4d9ce79d08df04dde718e53503444f5/asfsmd/core.py#L204-L218

File: eof/download.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SteffanDavies) [reply eof/download.py:100]: This was probably a typo when testing

**Pull request merged**